### ISSUE TITLE
Doubt regarding max_seq.

### ISSUE THREAD

**Issue by aditya-malte** [zihangdai/xlnet#114]
Hello,
As per the paper, XLNet caches the previous 'n' cells as memory.
My question is regarding max_seq_length:
1)Is it THE maximum length of sequence, as in are all subsequent tokens truncated?
2)Is it the previous 'n' cells(that I talked about) that are going to be cached. 
Also, how do we change(increase) the max_seq_len during inference because as I remember for the Transformer XL paper, the author(you) increase the number of cached states till it almost fills the GPU RAM during the inferences phase.
Thank You

Aditya Malte

**Comment by brendanxwhitaker** [comment]
The variable `max_seq_length` is the first of the two possibilities, the maximum length of a "sequence" as defined in the paper. Subsequent tokens are truncated, and shorter sequences are padded. 

`reuse_len` is the number of tokens cached for future use in subsequent computations. The current code doesn't appear to let you set it higher than `seq_len - 3` (see `data_utils.py`) during training, but you may be able to do this when making predictions. (Full disclosure: I haven't yet tried doing this.)

**Comment by aditya-malte** [comment]
3, that's it? So instead of just using 512 tokens it will use 515? I had expected cache size to be more like 256 or something like that.

**Comment by brendanxwhitaker** [comment]
It essentially is not allowing you to set `reuse_len` to be longer than `seq_len`. The `3` is just for padding purposes, due to the A/B split used for the BERT next sentence prediction objective (which XLNet does not use). There's a `[CLS]`, and two `[SEP]` tokens that get thrown in the training examples. The default/recommended `reuse_len` appears to be half of `seq_len`. However I'm not completely certain. Other input welcome.

**Comment by aditya-malte** [comment]
But where does the Transformer XL backbone come into picture then?

**Comment by brendanxwhitaker** [comment]
Good point, I looked into it and edited my answer above. It looks like the right variable is `mem_len`.
